packages/beeq/src/components/drawer/bq-drawer.tsx

```diff
@@ -0,0 +1,250 @@
+    if (!this.open) return;
+    if (!this.drawerElem || this.disableCloseEscKeydown || !(event.key === 'Escape' || event.key === 'Esc')) return;
+
+    await this.handleHide();
+  }
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  /** Method to be called to hide the drawer component */
+  @Method()
+  async hide(): Promise<void> {
+    await this.handleHide();
+  }
+
+  /** Method to be called to show the drawer component */
+  @Method()
+  async show(): Promise<void> {
+    await this.handleShow();
+  }
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private handleFooterSlotChange = () => {
+    this.hasFooter = hasSlotContent(this.footerElem, 'footer');
+  };
+
+  private handleHide = async () => {
+    const ev = this.bqClose.emit(this.el);
+    if (!ev.defaultPrevented) {
+      await leave(this.drawerElem);
+      this.open = false;
+      this.handleTransitionEnd();
+    }
+  };
+
+  private handleShow = async () => {
+    const ev = this.bqOpen.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.open = true;
+      await enter(this.drawerElem);
+      this.handleTransitionEnd();
+    }
+  };
+
+  private handleTransitionEnd = () => {
+    if (this.open) {
+      this.bqAfterOpen.emit();
+      return;
+    }
+    this.bqAfterClose.emit();
+  };
+
+  private getMoveTranslate = (): string => {
+    switch (this.placement) {
+      case 'right':
+        return `translate-x-full`;
+      case 'left':
+        return `-translate-x-full`;
+      default:
+        return '';
+    }
+  };
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <Host
+        class={{ 'is-hidden': !this.open }}
+        aria-hidden={!this.open ? 'true' : 'false'}
+        hidden={!this.open ? 'true' : 'false'}
+        role="dialog"
+      >
+        <div class={{ 'bq-drawer-backdrop': this.enableBackdrop }}></div>
+        <div
+          class={{ [`bq-drawer ${this.placement}`]: true }}
+          data-transition-enter="transition-all ease-out duration-300"
+          data-transition-enter-start={`opacity-0 ${this.getMoveTranslate()}`}
+          data-transition-enter-end="opacity-100"
+          data-transition-leave="transition-all ease-in duration-300"
+          data-transition-leave-start="opacity-100"
+          data-transition-leave-end={`opacity-0 ${this.getMoveTranslate()}`}
+          ref={(div) => (this.drawerElem = div)}
+          part="wrapper"
+        >
+          <main class="bq-drawer__content" part="content">
+            <header class="bq-drawer__header" part="header">
```

Comment:
Something seems to be off here, you are using `header` inside `main` when normally it should be:
```html
<header>...</header>
<main>...</main>
```